frontend/components/Blocks/CardsBlock/CardBlock.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from "react";
+import Card from "./Card";
+
+type Props = {
+  data: {
+    type: string;
+    value: {
+      cards: Array<CardItem>;
+    };
+    id: string;
+  };
+};
+
+type CardItem = {
+  title: string;
+  image_selector: {
+    id: number;
+    title: string;
+    img: {
+      src: string;
+      width: number;
+      height: number;
+      alt: string;
+    };
+  };
+  text: string;
+  card_background: string;
+};
+
+export default function CardBlock({ data }: Props) {
+  const [cardItems, setCardItems] = useState<Array<CardItem>>([]);
+
+  useEffect(() => {
+    const cardArray: Array<CardItem> = [];
+    if (data.value.cards.length) {
+      {
+        data.value?.cards.map((cardItem: CardItem) => {
+          cardArray.push(cardItem);
+        });
+      }
+    }
+    setCardItems(cardArray);
+  }, [data]);
+
+  return (
+    <div className={`flex flex-row w-full justify-center h-fit py-12 px-10 lg:px-16 lg:pt-16`}>
+      <div className={`flex flex-row  flex-wrap`}>
+        {cardItems.map((cardItem, _index) => {
+          return <Card cardItem={cardItem} key={_index}></Card>;
```

Comment:
This can be simplified by destructuring `card` from the props. `useState` and `useEffect` aren't needed.

Also keep in mind that prefixing a variable with an `_` is a way of communicating to readers that you don't intend on using the variable, but are keeping it to communicate intent. `_index` should therefore be `index`, since you *do* use it.

```suggestion
export default function CardBlock({ data: { value: { cards }} }: Props) {
  return (
    <div className={`flex flex-row w-full justify-center h-fit py-12 px-10 lg:px-16 lg:pt-16`}>
      <div className={`flex flex-row  flex-wrap`}>
        {cards.map((cardItem, index) => {
          return <Card cardItem={cardItem} key={index}></Card>;
```

Reply:
Thanks so much for this. I've removed the useState and useEffect and did this also in a few other cases.